src/IK/src/JointsTrackingTask.cpp

```diff
@@ -0,0 +1,159 @@
+    {
+        log()->error("{} KinDynComputations object is not valid.", errorPrefix);
+        return false;
+    }
+
+    auto ptr = paramHandler.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} The parameter handler is not valid.", errorPrefix);
+        return false;
+    }
+
+    if (!ptr->getParameter("robot_velocity_variable_name", m_robotVelocityVariableName))
+    {
+        log()->error("{} Error while retrieving the robot velocity variable.", errorPrefix);
+        return false;
+    }
+
+    // set the gains for the controllers
+    m_kp.resize(m_kinDyn->getNrOfDegreesOfFreedom());
+    if (!ptr->getParameter("kp", m_kp))
+    {
+        log()->error("{} Error while retrieving the proportional gain.", errorPrefix);
+        return false;
+    }
+
+    // set the description
+    m_description = "Joint tracking task";
+
+    m_zero = Eigen::VectorXd::Zero(m_kinDyn->getNrOfDegreesOfFreedom());
+    m_desiredJointPosition = Eigen::VectorXd::Zero(m_kinDyn->getNrOfDegreesOfFreedom());
+    m_desiredJointVelocity = Eigen::VectorXd::Zero(m_kinDyn->getNrOfDegreesOfFreedom());
+    m_jointPosition = Eigen::VectorXd::Zero(m_kinDyn->getNrOfDegreesOfFreedom());
+
+    m_isInitialized = true;
+
+    return true;
+}
+
+bool JointsTrackingTask::update()
+{
+    constexpr auto errorPrefix = "[JointsTrackingTask::update]";
+
+    if (!m_kinDyn->getJointPos(m_jointPosition))
+    {
+        log()->error("{} Unable to get the joint position.", errorPrefix);
+        return false;
+    }
+
+    m_b = m_desiredJointVelocity + m_kp.asDiagonal() * (m_desiredJointPosition - m_jointPosition);
+
+    return true;
+}
+
+bool JointsTrackingTask::setSetpoint(Eigen::Ref<const Eigen::VectorXd> jointPosition)
+{
+    return this->setSetpoint(jointPosition, m_zero);
+}
+
+bool JointsTrackingTask::setSetpoint(Eigen::Ref<const Eigen::VectorXd> jointPosition,
+                                     Eigen::Ref<const Eigen::VectorXd> jointVelocity)
+{
+    constexpr auto errorPrefix = "[JointsTrackingTask::setSetpoint]";
+
+    if (jointPosition.size() != m_kinDyn->getNrOfDegreesOfFreedom()
+        || jointVelocity.size() != m_kinDyn->getNrOfDegreesOfFreedom())
+    {
+        log()->error("{} Wrong size of the desired reference setpoint. Expected size: {}. Given "
+                     "joint position size: {}, given joint velocity size: {}.",
+                     errorPrefix,
+                     m_kinDyn->getNrOfDegreesOfFreedom(),
+                     jointPosition.size(),
+                     jointVelocity.size());
+        return false;
+    }
+
+    m_desiredJointPosition = jointPosition;
+    m_desiredJointVelocity = jointVelocity;
+
+    return true;
+}
+
+std::size_t JointsTrackingTask::size() const
+{
+    if (m_kinDyn == nullptr)
+    {
+        log()->warn("[JointsTrackingTask::size] Please call setKinDyn method before. A size equal "
+                    "to zero will be returned.");
+        return 0;
+    }
```

Comment:
This might be a good place for an ``assert``, in case the warning message gets unnoticed.

Reply:
Done in dd0018fdec02bc1cf173cbd88b2c581f9335de6e